lib/ConsumerImpl.cc

```diff
@@ -319,6 +320,45 @@ void ConsumerImpl::unsubscribeAsync(ResultCallback originalCallback) {
     }
 }
 
+void ConsumerImpl::triggerCheckExpiredChunkedTimer() {
+    checkExpiredChunkedTimer_->expires_from_now(
+        boost::posix_time::milliseconds(expireTimeOfIncompleteChunkedMessageMs_));
+    std::weak_ptr<ConsumerImplBase> weakSelf{shared_from_this()};
+    checkExpiredChunkedTimer_->async_wait([this, weakSelf](const boost::system::error_code& ec) -> void {
+        auto self = weakSelf.lock();
+        if (!self) {
+            return;
+        }
+        if (ec) {
+            LOG_DEBUG(getName() << " Check expired chunked messages was failed or cancelled, code[" << ec
+                                << "].");
+            return;
+        }
+        Lock lock(chunkProcessMutex_);
+        long currentTimeMs = TimeUtils::currentTimeMillis();
+        chunkedMessageCache_.removeOldestValuesIf(
+            [this, &currentTimeMs](const std::string& uuid, const ChunkedMessageCtx& ctx) -> bool {
```

Comment:
It's better capture `currentTimeMs` by value. Though the lambda is called before `removeOldestValuesIf` finished, capturing a reference (the 8 bytes underlying pointer) is not better than capturing a long (8 bytes).

Reply:
+1

Reply:
Question: if we use x86 architectures 32-bit, it will use a 32-bit(4 bytes) pointer here, right?

Reply:
IIRC, long is also 4 bytes in 32-bit arch. BTW, 4 bytes vs 8 bytes is not so much different.

Reply:
https://godbolt.org/z/eezdxMGnc @Demogorgon314